frontend/src/metabase/lib/security.js

```diff
@@ -0,0 +1,38 @@
+import passwordGenerator from "password-generator";
+import MetabaseSettings from "metabase/lib/settings";
+
+// generate a password that satisfies `complexity` requirements, by default the ones that come back in the
+// `password-complexity` Setting; must be a map like {total: 6, number: 1}
+export const generatePassword = complexity => {
+  complexity =
+    complexity || MetabaseSettings.passwordComplexityRequirements() || {};
+  // generated password must be at least `complexity.total`, but can be longer
+  // so hard code a minimum of 14
+  const len = Math.max(complexity.total || 0, 14);
+
+  let password = "";
+  let tries = 0;
+  while (!isStrongEnough(password) && tries < 100) {
```

Comment:
This code looks questionable but I just moved it without changes so that `password-generator` is not trying to slip into the static viz bundle